01-shapefile-attributes.Rmd

```diff
@@ -75,182 +79,292 @@ CSV to Shapefile in R</a>
+extent(aoiBoundary_HARV)
+extent(lines_HARV)
+extent(point_HARV)
 
 ```
 
-#Querying Shapefile Attributes
+When we type in the full file name we see the same three types of metadata with
+some additional attribute information.  
+
+```{r view-shapefile-metadata2 }
+
+aoiBoundary_HARV
+lines_HARV
+point_HARV
+
+```
+
+#Query Shapefile Attributes
 Shapefiles often contain an associated database or spreadsheet of values called
 `attributes` that describe the vector features in the shapefile. You can think
 of this like a spreadsheet with rows and columns. Each column in the spreadsheet
-is an individual `Attribute` that describes an object. Shapefile `attributes` 
-include measurements that correspond to the geometry of the shapefile features.
+is an individual `attribute` that describes an object (sometimes called
+variables). Shapefile `attributes` include measurements that correspond to the
+geometry of the shapefile features.
+
+For example, the Roads shapefile (`lines_HARV` object) contains an attribute
+called `TYPE`. Each line in the shapefile has an associated TYPE which describes
+the type of road (road, footpath, boardwalk, etc).  
 
-For example, the `roads` shapefile contains an attribute called `TYPE`. Each
-line in the shapefile has an associated type which describes the type of road, or 
-element in the file. 
+We can look at all of the associated data attributes by printing the contents of
+the `data` slot with `@data`. 
 
+```{r shapefile-attributes}
+#just view the attributes of the data
+(lines_HARV@data)
+```
+
+Bonus: If you are comfortable using `slot()`, it is another method to do the
+same thing. {: .notice1}
 
-You can quickly view all attributes of a spatial object in r using the `@data`. 
-Bonus: If you are comfortable using `slot()` it is another method to do the same
-thing. 
+Using `@data` we can see a table with attribute names across the top and rows
+for each line (or point/ polygon if using those) in the shapefile. The 0-12
+printed in the first column is simply a `row.name`. We can test this by issuing
+the following command to the `R` console: `row.names(aoiBoundary@data)`.
+
+``` {r row-names}
+#What are the row names?
+row.names(lines_HARV@data)
+```
+
+If we want to select something about the attribuates, we can do this by 
+combining common functions (e.g. `names` or `head`) and then using `NAME@data`.
 
 Let's give it a try.
 
 ```{r view-shapefile-attributes }
-
 #view just the attribute names for the lines spatial object
-names(lines@data)
-
-#view all attribute data for the lines spatial object
-lines@data 
+names(lines_HARV@data)
 
 #view just the top 5 attributes for the lines spatial object
-head(lines@data)
+head(lines_HARV@data)
 
-#view attributes for the other spatial objects
-aoiBoundary@data
-point@data
+```
+
+Alternatively, you can find attributes by using the `attributes` function,
+although if you have lots of objects in your shapefile this can be difficult to
```

Comment:
be more specific - not sure what this means? 
If you have lots of objects, the attributes function (or is it a method??) returns all of the ...